tests/test_bundle.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+import shlex
+from pathlib import Path
+from pytest_operator.plugin import OpsTest
+import lightkube
+from lightkube.resources.core_v1 import Service
+
+MODEL_NAME = "kubeflow"
+USERNAME = "admin"
+PASSWORD = "foobar"
+
+
+def get_ingress_ip(lightkube_client):
+    gateway_svc = lightkube_client.get(
+        Service, "istio-ingressgateway-workload", namespace=MODEL_NAME
+    )
+
+    endpoint = gateway_svc.status.loadBalancer.ingress[0].ip
+    return endpoint
+
+
+@pytest.fixture(scope="session")
+def lightkube_client():
+    yield lightkube.Client()
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_deploy(ops_test, request, lightkube_client):
+    if ops_test.model_name != MODEL_NAME:
+        raise ValueError("kfp must be deployed to namespace kubeflow")
+
+    bundle_file = request.config.getoption("bundle-file", default=None)
+    channel = request.config.getoption("channel", default=None)
+
+    if (bundle_file is None and channel is None) or (bundle_file and channel):
+        raise ValueError("One of --bundle-file or --channel is required")
+
+    model = ops_test.model_full_name
+
+    if bundle_file:
+        bundle_file_path = Path(bundle_file)
+        # deploy from file
+        cmd = f"juju deploy -m {model} --trust {bundle_file_path}"
+    if channel:
+        # deploy from charmhub
+        cmd = f"juju deploy kubeflow -m {model} --trust --channel {channel}"
```

Comment:
According to this, if both options are set we prioritize the channel option, is this intentional?

Reply:
I have a check a few lines above to make sure only one option is set. It would error out both or none are provided